### DISCUSSION TITLE
ERROR

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1359] [Q&A] [answered]
is there a possibility that errors such as the ones in this log: https://pastelink.net/y2kw70hn
only appear with the latest nightly build?

**Comment by kagikn** [comment, marked as answer]
Rename scripts like `ForestShelter.2.cs` and try again. Format is like: `[some name].[API version with single digit].cs`
SHVDN compiles raw script source code with a **non-deprecated** version, which is v3 only as of writing, since af0b731 (used first version without filtering before).

**Reply by Michael21107** [reply]
im sorry, u want me to add the `.3` to the name?

**Reply by kagikn** [reply]
yes, you need to. You could specify an API version for each script source code since [v3.1.0](https://github.com/scripthookvdotnet/scripthookvdotnet/releases/tag/v3.1.0) tho.

**Reply by kagikn** [reply]
Oops, I told you to add the wrong annotation, you need to add `.2` before `.cs`.

**Reply by Michael21107** [reply]
even if its a standalone script? eg: forestshelter.cs in scripts folder?

**Reply by kagikn** [reply]
you would need to rename every script that aren't compiled (`.cs` and `.vb` files that you can view as text files). For example, you won't need to rename `SomeScript.dll`, but you would need to rename `SomeScript.cs` to `SomeScript.2.cs` because it should be expected to use the v2 API. Otherwise it would have been named like `SomeSxript.3.cs` to explicitly specify an API version.

By the way, you can't directly reference to another script or a library that SHVDN doesn't automatically add as a reference without building source code into a compiled assembly.

**Reply by Michael21107** [reply]
so all .cs scripts are considered depracated (using V2)?

**Reply by kagikn** [reply]
.cs scripts themselves are not considered depracated, but I haven't seen any of them with the `.3` annotation, which will tells SHVDN to use the v3 API.

**Reply by Michael21107** [reply]
so what about this? what does the author have to do to avoid the errors with the latest CI build and what API version is it using?
[InteriorsAdditions.zip](https://github.com/scripthookvdotnet/scripthookvdotnet/files/13625026/InteriorsAdditions.zip) (I've got permission to use it for this matter)

**Reply by kagikn** [reply]
The author doesn't have to rename scripts, but users would have to rename if the author doesn't explicitly specify which API version to use. For the script you provided, the author could rename the script to `InteriorsAdditions.2.cs` so users won't have to rename.

**Reply by Michael21107** [reply]
so the script is using V2?

**Reply by kagikn** [reply]
the script is supposed to use v2, but SHVDN make it use v3 in the latest nightly/CI build if you don't rename.

**Reply by Michael21107** [reply]
gotcha, thanks
